test/unit/Faucet.ts

```diff
@@ -0,0 +1,245 @@
+    });
+
+    it("sets the pool as eligible", async () => {
+      await faucet.setPoolEligible(fakePool3.address, true);
+
+      const result = await faucet.isPoolEligible(fakePool3.address);
+
+      expect(result).equal(true);
+    });
+
+    it("sets the pool as uneligible", async () => {
+      await faucet.setPoolEligible(fakePool2.address, false);
+
+      const result = await faucet.isPoolEligible(fakePool2.address);
+
+      expect(result).equal(false);
+    });
+  });
+
+  describe("getTokenBalance", async () => {
+    const balancePool1 = BigNumber.from(42);
+    const balancePool2 = BigNumber.from(3);
+
+    beforeEach(async () => {
+      fakePool1.balanceOf.returns(balancePool1);
+      fakePool2.balanceOf.returns(balancePool2);
+    });
+
+    it("returns the balance of a single token", async () => {
+      const result = await faucet.getTokenBalances([fakePool1.address]);
+
+      expect(result).eql([balancePool1]);
+    });
+
+    it("returns the aggregated balance of a multiple tokens", async () => {
+      const result = await faucet.getTokenBalances([
+        fakePool1.address,
+        fakePool2.address,
+      ]);
+
+      expect(result).eql([balancePool1, balancePool2]);
+    });
+  });
+
+  describe("deposit", async () => {
+    it("reverts when the address is neither and eligible pool nor a valid toucan contract", async () => {
+      fakeToucanContractRegistry.checkERC20.reset();
+      fakeToucanContractRegistry.checkERC20.returns(false);
+      await faucet.setPoolEligible(fakePool1.address, false);
+
+      await expect(faucet.deposit(fakePool3.address, 42)).revertedWith(
+        "Token rejected"
+      );
+    });
+
+    it("transfers the given amount of tokens from the sender to the faucet", async () => {
+      const transferredAmount = BigNumber.from(42);
+      await faucet.connect(alice).deposit(fakePool1.address, transferredAmount);
+
+      expect(fakePool1.transferFrom).calledOnceWith(
+        alice.address,
+        faucet.address,
+        transferredAmount
+      );
+    });
+
+    it("emits a Deposited event", async () => {
+      const transferredAmount = BigNumber.from(42);
+      await expect(
+        faucet.connect(alice).deposit(fakePool1.address, transferredAmount)
+      )
+        .to.emit(faucet, "Deposited")
+        .withArgs(fakePool1.address, transferredAmount);
+    });
+  });
+
+  describe("checkIfWithdrawalTimeout", async () => {
+    it("returns false if the caller hasn't withdrawn", async () => {
+      const result = await faucet.checkIfWithdrawalTimeout();
+
+      expect(result).equal(false);
+    });
+
+    it("returns false if the caller withrew more than 30 seconds ago", async () => {
+      await faucet.withdraw(fakePool1.address, 42);
+      const currentTimestamp = (await ethers.provider.getBlock("latest"))
+        .timestamp;
+      network.provider.send("evm_setNextBlockTimestamp", [
+        currentTimestamp + 31,
```

Comment:
Read `TIMEOUT_LIMIT` and add 1 to it to avoid hardcoding the current const here.